pkg/cloud/services/gc/cleanup.go

```diff
@@ -71,21 +70,19 @@ func (s *Service) deleteResources(ctx context.Context) error {
 
 	cleanupFuncs := s.cleanupFuncs
 
-	if val, found := annotations.Get(s.scope.InfraCluster(), expinfrav1.ExternalResourceGCTasksAnnotation); found {
-		var gcTaskToFunc = map[string]ResourceCleanupFunc{
-			"load-balancer":  s.deleteLoadBalancers,
-			"target-group":   s.deleteTargetGroups,
-			"security-group": s.deleteSecurityGroups,
+	if val, found := annotations.Get(s.scope.InfraCluster(), infrav1.ExternalResourceGCTasksAnnotation); found {
+		var gcTaskToFunc = map[infrav1.GCTask]ResourceCleanupFunc{
+			infrav1.GCTaskLoadBalancer:  s.deleteLoadBalancers,
+			infrav1.GCTaskTargetGroup:   s.deleteTargetGroups,
+			infrav1.GCTaskSecurityGroup: s.deleteSecurityGroups,
 		}
 
 		cleanupFuncs = ResourceCleanupFuncs{}
 
 		tasks := strings.Split(val, ",")
 
-		// TODO: add some validation here.
```

Comment:
Did the TODO get resolved?

Reply:
Yes, by this PR

Reply:
Oh, I thought that the validation is meant to signify some kind of other validation here. :)